front/components/app/blocks/WebScrape.jsx

```diff
@@ -0,0 +1,160 @@
+    block.spec.url.startsWith("https://")
+      ? block.spec.url.slice("https://".length)
+      : block.spec.url
+  );
+  let { providers, isProvidersLoading, isProvidersError } = readOnly
+    ? {
+        providers: [],
+        isProvidersLoading: false,
+        isProvidersError: false,
+      }
+    : useProviders();
+
+  let browserlessAPIProvider = providers.find(
+    (p) => p.providerId == "browserlessapi"
+  );
+
+  // Update the config to impact run state based on the serpAPI provider presence.
+  if (!readOnly && !isProvidersLoading && !isProvidersError) {
+    if (
+      (!block.config.provider_id || block.config.provider_id.length == 0) &&
+      browserlessAPIProvider
+    ) {
+      setTimeout(() => {
+        let b = shallowBlockClone(block);
+        b.config.provider_id = "browserlessapi";
+        onBlockUpdate(b);
+      });
+    }
+    if (
+      block.config.provider_id &&
+      block.config.provider_id.length > 0 &&
+      !browserlessAPIProvider
+    ) {
+      setTimeout(() => {
+        let b = shallowBlockClone(block);
+        b.config.provider_id = "";
+        onBlockUpdate(b);
+      });
+    }
+  }
+
+  const handleUrlChange = (url) => {
+    let b = shallowBlockClone(block);
+    b.spec.url = url ? "https://" + url : "";
+    setUrlWithoutScheme(url);
+    onBlockUpdate(b);
+  };
+
+  const handleSelectorChange = (selector) => {
+    let b = shallowBlockClone(block);
+    b.spec.selector = selector;
+    onBlockUpdate(b);
+  };
+
+  return (
+    <Block
+      user={user}
+      app={app}
+      block={block}
+      status={status}
+      running={running}
+      readOnly={readOnly}
+      canUseCache={true}
+      onBlockUpdate={onBlockUpdate}
+      onBlockDelete={onBlockDelete}
+      onBlockUp={onBlockUp}
+      onBlockDown={onBlockDown}
+    >
+      <div className="flex flex-col mx-4 w-full">
+        <div className="flex flex-col space-y-1 text-sm font-medium text-gray-700 leading-8">
+          <div className="flex-initial flex flex-row items-center space-x-1">
+            {!isProvidersLoading && !browserlessAPIProvider && !readOnly ? (
+              <div className="px-2">
+                <Link href={`/${user}/providers`}>
+                  <a
+                    className={classNames(
+                      "inline-flex items-center rounded-md py-1 text-sm font-normal",
+                      "border px-3",
+                      readOnly
+                        ? "text-gray-300 border-white"
+                        : "text-gray-700 border-orange-400",
+                      "focus:outline-none focus:ring-0"
+                    )}
+                  >
+                    Setup Browserless API
+                  </a>
+                </Link>
+              </div>
+            ) : null}
+          </div>
+          <div className="flex w-full font-normal">
+            <div className="flex rounded-md flex-1">
+              Scrape this URL:&nbsp;
```

Comment:
url

Reply:
Is there a reason HTML and CSS are capitalized but URL isn't? I don't care either way, just curious.

Reply:
Not really, we can go for URL :+1: